OrbitLinuxTracing/UprobesReturnAddressManager.h

```diff
@@ -59,6 +59,34 @@ class UprobesReturnAddressManager {
     }
   }
 
+  void PatchCallchain(pid_t tid, uint64_t* callchain, uint64_t nr,
+                      unwindstack::Maps* maps) {
+    if (!tid_uprobes_stacks_.contains(tid)) {
+      return;
+    }
+
+    auto& tid_uprobes_stack = tid_uprobes_stacks_.at(tid);
+    CHECK(!tid_uprobes_stack.empty());
+
+    uint64_t i = 0;
+    // TODO(kuebler): What about tail-call optimization, where two uretprobes
```

Comment:
Have you tried what happens? Maybe in such case the second uretprobe (the first to be encountered in the loop) shouldn't be applied? In other words, maybe when there are consecutive `OpenUprobes`s with the same stack pointer, only the outer one should be applied (just guessing here)? A basic solution might end up being quite easy.

Reply:
Fine, but don't consider this a rare corner case, tail call optimization is common.

Reply:
Right, I'll add a bug for this and will come back as to this as soon as possible. For now, need to re-prioritize.